libs/framework/CMakeLists.txt

```diff
@@ -66,6 +66,7 @@ install(DIRECTORY include/ DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}/celix/framewo
 install(DIRECTORY include_deprecated/ DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}/celix/framework COMPONENT framework)
 install(DIRECTORY ${CMAKE_BINARY_DIR}/celix/gen/includes/framework/ DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}/celix/framework COMPONENT framework)
 
+celix_target_hide_symbols(framework)
```

Comment:
I think celix_target_hide_symbols is more explicit, but this was already done with the target properties (only for C visiblity, but most libs are only C). 

See https://github.com/apache/celix/pull/584/files#diff-556b07494eb3b34b34072320fa4a8e72403f81725c9d7d7b35fc6e4243fd2b06R44

This is also the case for utils, etcdlib, dfi.

Reply:
Ouch!